src/Input/Select/Select.test.tsx

```diff
@@ -360,3 +360,196 @@ it('up and down arrow keys can be used to navigate the options menu', () => {
+    const onInputChangeSpy = jest.fn();
+    const Component = () => (
+      <React.Fragment>
+        <Select label="onInputChange" onInputChange={onInputChangeSpy}>
+          <Select.Option value="onInputChange">onInputChange</Select.Option>
+        </Select>
+      </React.Fragment>
+    );
+
+    const mockValue = 'mock value';
+    const { queryByRole } = render(<Component />);
+    const selectInput = queryByRole('combobox');
+    fireEvent.change(selectInput, {
+      target: {
+        value: mockValue,
+      },
+    });
+    expect(onInputChangeSpy).toHaveBeenCalledTimes(1);
+    // not able to validate SyntheticEvent
+  });
+});
+
+describe('<Select/> disableTyping', () => {
+  it('esc key should have no effect', async () => {
+    const Component = () => (
+      <React.Fragment>
+        <Select label="disableTyping" disableTyping={true}>
+          <Select.Option value="disableTyping">disableTyping</Select.Option>
+        </Select>
+      </React.Fragment>
+    );
+
+    const { queryByRole, queryByTestId } = render(<Component />);
+    const selectInput = queryByRole('combobox');
+    const selectList = queryByTestId('listbox');
+
+    fireEvent.focus(selectInput);
+    expect(selectList.hasAttribute('open')).toEqual(true);
+    fireEvent.keyDown(selectInput, { key: 'Esc', keyCode: 27 });
+    expect(selectList.hasAttribute('open')).toEqual(true);
+  });
+});
+
+describe('<Select/> esc key', () => {
+  it('esk keydown event should close the option list', async () => {
+    const Component = () => (
+      <React.Fragment>
+        <Select label="disableTyping">
+          <Select.Option value="disableTyping">disableTyping</Select.Option>
+        </Select>
+      </React.Fragment>
+    );
+
+    const { queryByRole, queryByTestId } = render(<Component />);
+    const selectInput = queryByRole('combobox');
+    const selectList = queryByTestId('listbox');
+
+    fireEvent.focus(selectInput);
+    expect(selectList.hasAttribute('open')).toEqual(true);
+    fireEvent.keyDown(selectInput, { key: 'Esc', keyCode: 27 });
+    expect(selectList.hasAttribute('open')).toEqual(false);
+  });
+});
+
+describe('<Select/> handleMouseEnterOption', () => {
+  it('it should match snapshot, the option element should have active class name', async () => {
+    const Component = () => (
+      <React.Fragment>
+        <Select label="handleMouseEnterOption">
+          <Select.Option value="handleMouseEnterOption">
+            handleMouseEnterOption
+          </Select.Option>
+        </Select>
+      </React.Fragment>
+    );
+
+    const { queryByRole, queryByTestId, asFragment } = render(<Component />);
+    const selectInput = queryByRole('combobox');
+    const option = queryByTestId('option');
+    const selectList = queryByTestId('listbox');
+
+    fireEvent.focus(selectInput);
+    expect(selectList.hasAttribute('open')).toEqual(true);
+    fireEvent.mouseEnter(option);
+    expect(asFragment()).toMatchSnapshot();
```

Comment:
I think we should have `expect(option).toHaveClass('active');`

Reply:
please check snapshot the element has class `active`